src/bootstrap/builder.rs

```diff
@@ -938,6 +938,12 @@ impl<'a> Builder<'a> {
         // but this breaks CI. At the very least, stage0 `rustdoc` needs `--cfg bootstrap`. See
         // #71458.
         let mut rustdocflags = rustflags.clone();
+        rustdocflags.env("RUSTDOCFLAGS");
```

Comment:
This seems different from the handling in Rustflags::new, which [also checks the cargo environment variable](https://github.com/estebank/rust/blob/8ead8f57c6bd57e327ef150ac99e47a4c2486000/src/bootstrap/builder.rs#L1565) - we might want to do that here too? I think it might be worth a function we can call with the "prefix" of the env variables (rustdoc vs rustc) and it can do the rest to properly load things up?

Reply:
Good catch, thanks. I added a helper function and used it for both `rustflags` and `rustdocflags`.